packages/flutter/lib/src/rendering/unconstrained_box.dart

```diff
@@ -0,0 +1,115 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
```

Comment:
I'd probably just put this in shifted_box.dart, along with all the other RenderShiftedBox subclasses. It's a bit weird to have just this class be special and have its own file. (Maybe all the RenderAligningShiftedBoxes should be in their own file? Or maybe each class should have its own file? I don't really have a strong opinion here, so long as we can explain the rule as to where a class should go to the next person...)

Reply:
Yeah we use a separate file for the elaborate RenderObjects with long layout functions (typically, those with more than one child).